pkg/pwalk/pwalk.go

```diff
@@ -29,16 +29,24 @@ type WalkFunc = filepath.WalkFunc
 // - if more than one walkFn instance will return an error, only one
 // of such errors will be propagated and returned by Walk, others
 // will be silently discarded.
+//
+// Deprecated: use github.com/opencontainers/selinux/pwalkdir.Walk
 func Walk(root string, walkFn WalkFunc) error {
-	return WalkN(root, walkFn, runtime.NumCPU()*2)
+	return walkN(root, walkFn, runtime.NumCPU()*2)
 }
 
 // WalkN is a wrapper for filepath.Walk which can call multiple walkFn
 // in parallel, allowing to handle each item concurrently. A maximum of
 // num walkFn will be called at any one time.
 //
 // Please see Walk documentation for caveats of using this function.
+//
+// Deprecated: use github.com/opencontainers/selinux/pwalkdir.WalkN
 func WalkN(root string, walkFn WalkFunc, num int) error {
+	return walkN(root, walkFn, num)
+}
+
+func walkN(root string, walkFn WalkFunc, num int) error {
```

Comment:
What is the meaning of this change? Is that for one deprecated function to not use another? In this case perhaps it's easier to annotate instead.

Reply:
Yeah, both my IDE and linter were complaining about using deprecated functions; so could've either added a `//nolint:<xxx>` flag, or do this.


I'm wondering though if we perhaps should immediately proceed with aliasing this to the new package, if we're going to support go1.16+ (https://github.com/opencontainers/selinux/pull/183), WDYT?

Reply:
I changed this back.

w.r.t. aliasing; we can't alias it as the signatures don't match, so let's deprecate it as step 1, then remove after